lib/sensu/client/http_socket.rb

```diff
@@ -0,0 +1,208 @@
+              "POST" => method(:process_request_results)
+            },
+            "help" => "Send check json results here"
+          },
+          "/settings" => {
+            "methods" => {
+              "GET" => method(:process_request_settings)
+            },
+            "help" => "Get sensu settings (requires basic auth)"
+          },
+          "/brew" => {
+            "methods" => {
+              "GET" => Proc.new { |response|
+                response.status = 418
+                response.status_string = "I'm a teapot"
+                response.content = Sensu::JSON::dump({:response => "I'm a teapot!"})
+                response.send_response
+              }
+            },
+            "help" => "Ask Sensu to brew a cup of joe (try it!)"
+          }
+        }
+      end
+
+      def authorized?
+        http_options = @settings[:client][:http_socket] || Hash.new
+        if http_options[:user] and http_options[:password]
+          if @http[:authorization]
+            scheme, base64 = @http[:authorization].split("\s")
+            if scheme == "Basic"
+              user, password = Base64.decode64(base64).split(":")
+              return (user == http_options[:user] && password == http_options[:password])
+            end
+          end
+        end
+        false
+      end
+
+      def process_request_info(response)
+        @logger.debug("Processing info request")
+        transport_info do |info|
+          rdata = {
+            :sensu => {
+              :version => VERSION
+            },
+            :transport => info
+          }
+          response.content = Sensu::JSON::dump(rdata)
+          response.send_response
+        end
+      end
+
+      def process_request_results(response)
+        @logger.debug("Processing results request")
+        if @http[:content_type] and @http[:content_type] == 'application/json' and @http_content
+          begin
+            check = Sensu::JSON::load(@http_content)
+            process_check_result(check)
+            rdata = {
+              :response => "ok"
+            }
+            response.content = Sensu::JSON::dump(rdata)
+            response.send_response
+          rescue Sensu::JSON::ParseError, ArgumentError
+            rdata = {
+              :response => "Failed to parse json body"
+            }
+            response.status = 400
+            response.status_string = "Failed to parse json body"
+            response.content = Sensu::JSON::dump(rdata)
+            response.send_response
+          end
+        else
+          response.status = 415
+          response.status_string = "Only json content type accepted"
+          rdata = {
+            :response => "Invalid content type"
+          }
+          response.content = Sensu::JSON::dump(rdata)
+          response.send_response
+        end
+      end
+
+      def process_request_settings(response)
+        if authorized?
+          @logger.debug("Processing settings request")
```

Comment:
yeah I added most of those when still confirming my routes were working, I think now it's just better to stick to the one log statement when processing starts, we're already logging the endpoint there